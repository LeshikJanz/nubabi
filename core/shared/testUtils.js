/* eslint-disable import/no-extraneous-dependencies */
import configureMockStore from 'redux-mock-store';
import { createEpicMiddleware } from 'redux-observable';
import firebaseMock from 'firebase-mock';

export const setupEpic = epic => {
  const epicMiddleware = createEpicMiddleware(epic);
  const mockStore = configureMockStore([epicMiddleware]);

  return {
    middleware: epicMiddleware,
    mockStore,
    afterEach: () => {
      epicMiddleware.replaceEpic(epic);
    },
  };
};

export const mockFirebase = () => {
  const mockDb = new firebaseMock.MockFirebase();
  const mockAuth = new firebaseMock.MockFirebase();
  const mockSdk = firebaseMock.MockFirebaseSdk(
    path => {
      return path ? mockDb.child(path) : mockDb;
    },
    () => {
      return mockAuth;
    },
  );

  return mockSdk;
};
