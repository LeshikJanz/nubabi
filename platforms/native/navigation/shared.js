import React from 'react';
import { pathOr } from 'ramda';
import LeftHeader from './LeftHeader';
import RightHeader from './RightHeader';

const babyNameTitle = ({ state }) =>
  pathOr('Nubabi', ['params', 'babyName'], state);

export function getBabyTitle(navigation) {
  if (!navigation) {
    throw new Error('an screen hasnt been updated, check stack');
  }

  return babyNameTitle(navigation);
}

export const getTabHeaders = navigate => ({
  left: <LeftHeader navigate={navigate} />,
  right: <RightHeader navigate={navigate} />,
});
