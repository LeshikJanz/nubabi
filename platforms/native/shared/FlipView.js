// @flow
import React, { Component } from 'react';
import { View, Animated } from 'react-native';

const styles = {
  animatedContainer: {
    flex: 1,
  },
};

type Props = {
  renderFront: Function,
  renderBack: Function,
  velocity: number,
  tension: number,
  friction: number,
};

export class FlipCard extends Component {
  props: Props;

  state = {
    animatedValue: new Animated.Value(0),
    isFlipped: true,
  };

  componentDidUpdate(prevProps: Props, prevState: *) {
    if (this.state.isFlipped !== prevState.isFlipped) {
      this._flippedCard();
    }
  }

  _flipToggleCard = () => {
    this.setState({ isFlipped: !this.state.isFlipped });
  };

  _flippedCard() {
    Animated.spring(this.state.animatedValue, {
      toValue: 0, // Returns to the start
      velocity: this.props.velocity, // Velocity makes it move
      tension: this.props.tension, // Slow
      friction: this.props.friction, // Oscillate a lot
    }).start();
  }

  flippedCardView(isFlipped) {
    if (isFlipped) {
      return this.props.renderFront(this._flipToggleCard);
    }
    return this.props.renderBack(this._flipToggleCard);
  }

  render() {
    const perspective = 2000;
    const rotateY = this.state.animatedValue.interpolate({
      inputRange: [0, 1],
      outputRange: ['0deg', '180deg'],
    });
    return (
      <View style={styles.animatedContainer}>
        <Animated.View
          style={[
            styles.animatedContainer,
            { transform: [{ rotateY }, { perspective }] },
          ]}
        >
          {this.flippedCardView(this.state.isFlipped)}
        </Animated.View>
      </View>
    );
  }
}

export default FlipCard;
