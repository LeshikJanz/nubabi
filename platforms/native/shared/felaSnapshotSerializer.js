// WIP
// prettier-ignore
const felaSnapshotSerializer = {
  test(val) {
    return val && val.felaRules && val.$$typeof === Symbol.for('react.test.json');
  },

  print(val, print) {
    const styles = val.felaRules;
    val.felaRules = false;

    return `style = Object ${JSON.stringify(styles, null, 4)}\n\n${print(val)}`;
  },
};

module.exports = felaSnapshotSerializer;
