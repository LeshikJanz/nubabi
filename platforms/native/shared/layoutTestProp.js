export default {
  fontScale: 2,
  height: 1334,
  scale: 2,
  width: 750,
  viewportWidth: 750,
  viewportHeight: 1334,
};
