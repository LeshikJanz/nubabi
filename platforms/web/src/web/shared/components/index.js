import About from "./About";
import AsyncComponent from "./AsyncComponent";
import AuthenticatedRoute from "./AuthenticatedRoute";
import Header from "./Header";
import Home from "./Home";
import Loader from "./Loader";
import Test from "./Test";

export default {
  About,
  AsyncComponent,
  AuthenticatedRoute,
  Header,
  Home,
  Loader,
  Test
};
