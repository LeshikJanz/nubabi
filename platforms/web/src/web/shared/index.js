import * as actions from "./actions";
export { default as containers } from "./containers";
export { default as reducer } from "./reducer";
export { default as epics } from "./epics";
export { actions };
