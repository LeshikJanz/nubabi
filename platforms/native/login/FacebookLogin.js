// @flow
import React, { PureComponent } from 'react';
import { TouchableOpacity } from 'react-native';
import { AccessToken, LoginManager } from 'react-native-fbsdk';
import { compose } from 'ramda';
import { connect } from 'react-redux';
import { appError } from 'core/app/actions';
import { loginRequest } from 'core/auth/actions';
import { Box, Icon, Text } from '../components';

type Props = {
  appError: typeof appError,
  loginRequest: typeof loginRequest,
};

export class FacebookLogin extends PureComponent<Props> {
  handleLogin = () => {
    /* eslint-disable no-shadow */
    const { appError, loginRequest } = this.props;
    LoginManager.logInWithReadPermissions(['public_profile', 'email']).then(
      result => {
        if (result.isCancelled) {
          appError(new Error('Facebook login was cancelled'));
        } else {
          AccessToken.getCurrentAccessToken()
            .then(data => {
              loginRequest(data, 'facebook');
            })
            .catch(error => appError(error));
        }
      },
      error => {
        appError(error);
      },
    );
    /* eslint-enable no-shadow */
  };

  render() {
    return (
      <Box marginVertical={2} justifyContent="center" alignItems="center">
        <Box>
          <Text color="white">OR</Text>
        </Box>
        <Box
          as={TouchableOpacity}
          onPress={this.handleLogin}
          flexDirection="row"
          alignItems="center"
          marginTop={1}
        >
          <Box
            alignItems="flex-start"
            justifyContent="flex-start"
            style={() => ({
              backgroundColor: 'white',
              marginTop: 5,
              padding: 0,
              margin: 0,
              height: 18,
            })}
          >
            <Icon
              name="logo-facebook"
              size={24}
              color="#3B5998"
              style={{
                backgroundColor: 'transparent',
                padding: 0,
                margin: 0,
                marginTop: -3,
                alignSelf: 'stretch',
              }}
            />
          </Box>
          <Text
            medium
            color="white"
            marginLeft={0.5}
            style={() => ({
              marginTop: 2,
            })}
          >
            Log In with Facebook
          </Text>
        </Box>
      </Box>
    );
  }
}

export default compose(
  connect(null, {
    appError,
    loginRequest,
  }),
)(FacebookLogin);
