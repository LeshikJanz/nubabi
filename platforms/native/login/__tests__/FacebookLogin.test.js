import 'react-native';
import React from 'react';
import { FacebookLogin } from '../FacebookLogin';
import { expectRender } from '../../shared/testUtils';

test('it renders correctly', () => {
  expectRender(<FacebookLogin />);
});
