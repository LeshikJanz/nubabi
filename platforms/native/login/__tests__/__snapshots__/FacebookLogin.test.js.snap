// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
style = Object {
    "marginBottom": 32,
    "marginTop": 2,
    "borderColor": "#9EABBC",
    "borderBottomWidth": 0,
    "borderLeftWidth": 0,
    "borderRightWidth": 0,
    "borderTopWidth": 0,
    "alignItems": "flex-start",
    "justifyContent": "flex-start",
    "fontSize": 13,
    "lineHeight": 16,
    "color": "#FFFFFF",
    "fontFamily": "SF Pro Text",
    "flexDirection": "row",
    "backgroundColor": "white",
    "padding": 0,
    "margin": 0,
    "height": 18,
    "marginLeft": 8,
    "fontWeight": "500"
}

<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomWidth": 0,
      "borderColor": "#9EABBC",
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "justifyContent": "center",
      "marginBottom": 32,
      "marginTop": 32,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#9EABBC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "borderBottomWidth": 0,
          "borderColor": "#9EABBC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "color": "#FFFFFF",
          "fontFamily": "SF Pro Text",
          "fontSize": 13,
          "lineHeight": 16,
        }
      }
    >
      OR
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0,
        "borderColor": "#9EABBC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "flexDirection": "row",
        "marginTop": 16,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "white",
          "borderBottomWidth": 0,
          "borderColor": "#9EABBC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "height": 18,
          "justifyContent": "flex-start",
          "margin": 0,
          "marginTop": 5,
          "padding": 0,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#3B5998",
              "fontSize": 24,
            },
            Object {
              "alignSelf": "stretch",
              "backgroundColor": "transparent",
              "margin": 0,
              "marginTop": -3,
              "padding": 0,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        ïˆ¶
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "borderBottomWidth": 0,
          "borderColor": "#9EABBC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "color": "#FFFFFF",
          "fontFamily": "SF Pro Text",
          "fontSize": 13,
          "fontWeight": "500",
          "lineHeight": 16,
          "marginLeft": 8,
          "marginTop": 2,
        }
      }
    >
      Log In with Facebook
    </Text>
  </View>
</View>
`;
